include/aws/http/connection.h

```diff
@@ -215,6 +215,24 @@ struct aws_http2_connection_options {
      * See `aws_http2_on_remote_settings_change_fn`.
      */
     aws_http2_on_remote_settings_change_fn *on_remote_settings_change;
+
+    /**
+     * Optional.
+     * Set to true to manually manage the flow-control window of whole HTTP/2 connection.
+     *
+     * If false, the connection will maintain its flow-control windows such that
+     * no back-pressure is applied and data arrives as fast as possible.
+     *
+     * If true, the flow-control window of the whole connection will shrink as body data
+     * is received (headers, padding, and other metadata do not affect the window) for every streams
```

Comment:
Padding [DOES](https://httpwg.org/specs/rfc7540.html#DATA) affect it

> The entire DATA frame payload is included in flow control, including the **Pad Length** and **Padding** fields if present


And I don't think our API offers any way for users to know how much padding they're received 😐

Reply:
🤦‍♂️Oh, shoot. Our comments is wrong all the time...
We shrink the windows with the whole length of the data frame, but users don't have anyway to know about the padding.....
So, I think we can offer:
- on_data_begin with the length of the whole payload?
- on_padding with the length of padding?
- We only shrink the length without padding (We send out window update for user when there is padding)? I don't like we do thing behind for user... But I think that's probably fine...